### DISCUSSION TITLE
Use Renovate on private registry with authent creds store in the audited projects

### DISCUSSION THREAD

**Discussion by Trevaxx** [renovatebot/renovate#19804] [Request Help] [answered]
Hi,

I use Renovate and I need to connect it to different private registries.
I manage this through the environment variable: RENOVATE_HOST_RULES.

The tool is on the perimeter of different projects that need access to different private registries. These credentials cannot be shared in the Renovate project (because it is sensitive).

Is there a way for renovate (via a maintainer token) to use/overwrite its RENOVATE_HOST_RULES variable with the one defined in the audited project (if any).

**Comment by rarkins** [comment, marked as answer]
I'm not sure I understand the question, but you can configure `hostRules` in the repo config too, which would supersede any you define in the bot config using `RENOVATE_HOST_RULES`

**Reply by Trevaxx** [reply]
I did the tests and if I configure hostRules access to my private registries, in my audited projects (and not in the renovate orchestrator project), these authent informations are not taken into account by Renovate.

**Reply by rarkins** [reply]
I don't know what you mean by "audited projects", but I will say that `hostRules` in a `renovate.json` definitely should be taken into consideration

**Reply by Trevaxx** [reply]
Hello, my apologies for not being clearer the first time. I will try to explain beter.

# How are you running Renovate?
Self-hosted

# If you're self-hosting Renovate, tell us what version of Renovate you run.
Latest Docker Image

# If you're self-hosting Renovate, select which platform you are using.
gitlab.com

# Context
Renovate runs on 2 projects: Project A and Project B
I use dependencies from a private registry in my project A. The authentication information is secret and should not be managed at the Renovate project level but at the A project level.
I have potentially the same issues in my project B.

It is not possible for confidentiality reasons to centralize the 2 authentication information in the Renovate project.

So the authentication information must be managed in the A and B project and not in the Renovate project.
-> This is possible by configuring the "hostRules" in the renovate.json file of repos A and B

# Problem

Authentication information must be configured in the renovate.json file via "hostRules".
How to pass the username and password values ​​as an environment variable (do not write them in plain text).
Unlike inside the Renovate project, it doesn't seem like I can go through a config.js file do inject gitlab env variables.

**Reply by viceice** [reply]
you can configure encryption[^1] on bot, so you can encrypt your secrets inside repo and only the bot can decrypt at runtime.

[^1]: https://docs.renovatebot.com/getting-started/private-packages/#adminbot-config-vs-userrepository-config-for-self-hosted-users

**Reply by Trevaxx** [reply]
It works like a charm, thanks to both of you !
